### PR TITLE
Update CPU tolerance for f16 triplet margin loss

### PR DESCRIPTION
**Pull request by GeorgeWigley**
Currently, the `test_torchinductor_opinfo` test for `nn.functional.triplet_margin_loss` fails on AArch64, this PR increases the acceptable ATOL and RTOL for this test when using F16. There is precedent for this as XPU and CUDA already increase the tolerance. Additionally, the CPU backend increases the tolerance for the `with_distance_loss` variant of `triplet_margin_loss`. 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

### CODE REVIEW

**Review by leslie-fang-intel — APPROVED** [review]

**Comment by GeorgeWigley** [comment]
@leslie-fang-intel please could you merge this as I don't have permission?

**Comment by leslie-fang-intel** [comment]
I think I also doesn't have the permission for this file. cc @jansel

**Review by jansel — APPROVED** [review]
